app/services/course_serializers_service_v3.rb

```diff
@@ -0,0 +1,43 @@
+class CourseSerializersServiceV3
```

Comment:
just a passing thought, can this be a factory service or something, default is `v2`?

Reply:
Hoping @danielburnley might be able to advise on that

Reply:
How about including `CourseSerializersServiceV3` and changing what's different? Or use inheritance, I think either is a valid way to do code reuse here.

Reply:
Just for the benefit of anyone else having a look, here's the diff between V2 and V3 of the serialiser. I think it's worth trying to include/extend V2 into V3 and just overriding the methods below, but it'll probably take a little bit of fiddling to make sure `enrichment_attribute` and `meta` are over-ridden correctly.

```diff
modified   Users/misaka/Code/DfE/manage-courses-backend/app/serializers/api/v2/serializable_course.rb -> Users/misaka/Code/DfE/manage-courses-backend/app/serializers/api/v3/serializable_course.rb
@@ -1,10 +1,10 @@
 module API
-  module V2
+  module V3
     class SerializableCourse < JSONAPI::Serializable::Resource
       class << self
         def enrichment_attribute(name, enrichment_name = name)
           attribute name do
-            @object.enrichments.last&.__send__(enrichment_name)
+            @object.enrichments.select(&:published?).last&.__send__(enrichment_name)
           end
         end
       end
@@ -61,20 +61,6 @@ module API
       enrichment_attribute :personal_qualities
       enrichment_attribute :required_qualifications
       enrichment_attribute :salary_details
-
-      meta do
-        serializer_service = SerializeCourseService.new
-        edit_options = @object.edit_course_options
-
-        edit_options[:modern_languages] =
-          serializer_service.execute(object: edit_options[:modern_languages])[:serialized][:data]
-        edit_options[:subjects] =
-          serializer_service.execute(object: edit_options[:subjects])[:serialized][:data]
-
-        {
-          edit_options: edit_options,
-        }
-      end
     end
   end
 end
```

Reply:
Is there anything else we intend to remove from V3 after your investigation work @misaka ?

Reply:
Does extending V2 here increase the risk of inadvertently breaking V3 with a change to V2? I wouldn't expect that to happen if I was working on an explicitly versioned serializer. I would expect it to be isolated. Maybe split the common functionality into a separate module and include it in both so it's clearer the code is being used in more than one place?

Reply:
@gpeng That's a good point. This whole thing is a bit of a hack, since we just created the API endpoint and ran with it. I don't think v3 is the shape we want, there's a bunch of extra stuff in there that we don't want to be exposing, so I'm not sure what to do about that aside from what we're already doing ... vetting the attributes and deciding on what to keep/drop/change.

Reply:
I think the best case scenario is that we move to a stable API (v3.1 probably at this point), and then deprecate v3 and remove it as soon as Find and Apply aren't using it any more.

Maybe for now the answer is to duplicate, ship and move onto making v3.1

Reply:
So @misaka. Merge and Ship? :shipit: